mesonbuild/modules/python.py

```diff
@@ -297,6 +300,55 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def _get_install_scripts(self) -> T.List[mesonlib.ExecutableSerialisation]:
+        backend = self.interpreter.backend
+        ret = []
+        optlevel = self.interpreter.environment.coredata.get_option(mesonlib.OptionKey('bytecompile', module='python'))
+        if optlevel == -1:
+            return ret
+
+        installdata = backend.create_install_data()
+        py_files = []
+
+        def should_append(f, isdir: bool = False):
+            return f.startswith(('{py_platlib}', '{py_purelib}')) and (f.endswith('.py') or isdir)
+
+        for t in installdata.targets:
+            if should_append(t.out_name):
+                py_files.append((t.out_name, os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname))))
+        for d in installdata.data:
+            if should_append(d.install_path_name):
+                py_files.append((d.install_path_name, os.path.join(installdata.prefix, d.install_path)))
+        for d in installdata.install_subdirs:
+            if should_append(d.install_path_name, True):
+                py_files.append((d.install_path_name, os.path.join(installdata.prefix, d.install_path, '')))
+        pycompile = os.path.join(self.interpreter.environment.get_scratch_dir(), 'pycompile.py')
+        if os.path.exists(pycompile):
+            os.remove(pycompile)
+        for i in self.installations.values():
+            if isinstance(i, PythonExternalProgram) and i.run_bytecompile[i.info['version']]:
+                import importlib.resources
+
+                i = T.cast(PythonExternalProgram, i)
+                manifest = f'python-{i.info["version"]}-installed.json'
+                manifest_json = []
+                with open(pycompile, 'wb') as f:
+                    f.write(importlib.resources.read_binary('mesonbuild.scripts', 'pycompile.py'))
+                for name, f in py_files:
+                    if f.startswith((os.path.join(installdata.prefix, i.platlib), os.path.join(installdata.prefix, i.purelib))):
```

Comment:
Not sure this works on Windows if we don't normalize to posix path.

Reply:
I think it should, because we are comparing to `os.path.join(installdata.prefix, d.install_path)` above.

Reply:
Also we do test this functionality on cygwin and msys2, so it should work there. We do not test it on Azure Pipelines because VS tests don't support installation in the test harness.

Reply:
Unless I'm mistaken, it compares:
- `os.path.join(installdata.prefix, d.install_path)`
- `os.path.join(installdata.prefix, i.platlib)`

So that's assuming that `d.install_path` and `i.platlib` uses the same separator. I guess that's a pretty safe bet indeed.

Reply:
Yup.